csrc/scheduler/normalization_utils.h

```diff
@@ -311,6 +311,8 @@ int64_t getMaxRegOrSharedMemorySizeForPersistentBuffer(
 // inputs are cached instead of the persistent buffers. The decision of
 // projection is primarily based on the required sizes of the two cases --
 // projection is done if projecting to the inputs results in a smaller size.
+// If the scheduler is innerOuter with outer broadcast, projection is allowed
+// even it leads to a larger buffer size.
```

Comment:
Why?

Reply:
It is the logic exist in the current main branch. We need to re-evaluate this logic, for example if there is an extreme case where projecting to inputs increased the buffer size by 10 times, then we probabally don't want to project to inputs.
added comments:
```
// If the scheduler is innerOuter with outer broadcast, projection is allowed
// even it leads to a larger buffer size becuase the scheduled kernel allows the
// reuse of the outer broadcast Tv when iterating over the outer reduction
// dimension and leads to higher performance ( TODO: needs re-evaluate, may not
// true if the buffer size is increased a lot when projecting to inputs). See
// https://github.com/NVIDIA/Fuser/issues/402
```

Reply:
I'm still confused. If it's the current behavior, why does this function need to change?

Reply:
We are changing the logic.

This is the old logic:
> // ... The decision of 
// projection is primarily based on the required sizes of the two cases --
// projection is done if projecting to the inputs results in a smaller size.

The new code where Liqiang added a knob here that allowed projection to happen regardless of that restriction.
> // If the scheduler is innerOuter with outer broadcast, projection is allowed

Reply:
Thanks @jjsjann123.

Reply:
Please update the comment

Reply:
updated, looks like some of my local changes are not pushed to remote...